### ISSUE TITLE
Weak password complexity requirements

### ISSUE THREAD

**Issue by mprins** [flamingo-geocms/flamingo#1656]
### Version
Which version of flamingo do you use: 5.7.2-SNAPSHOT

### Where and/or What

viewer-admin

### What happened

- Passwords have a minimum required length of (only) 8 characters
- No complexity requirements are enforced when an admin add new users or changes a users password

### Reproduce
How can we reproduce this? 

  1.  add a new uses with password `flamingo`
  

related #1233

**Comment by mprins** [comment]
The relevant code:

https://github.com/flamingo-geocms/flamingo/blob/6ba016ffc19b3f6baa41e1b29d67caebe1a69857/viewer-admin/src/main/java/nl/b3p/viewer/admin/stripes/UserActionBean.java#L269-L275

Can easily be upgraded to a larger password length and a complexity check could be added there as well; though complexity requirements as such are no longer considererd a "best practice" [eg. https://www.pluralsight.com/blog/security-professional/modern-password-guidelines]

Something we could look into is using: https://github.com/GoSimpleLLC/nbvcxz

**Comment by mtoonen** [comment]
I wouldn't opt for complex passwords (weird signs, numbers, etc.). See also https://xkcd.com/936/

Longer passwords should be more secure, although 8 characters should be fine: https://en.wikipedia.org/wiki/Password_strength

**Comment by mprins** [comment]
Note that the current password policy (length/complexity) has received a score of "CVSS:3.0/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:N" of in a penetration test, flagging this as a "LOW" level vulnerabilty

see: https://en.wikipedia.org/wiki/Common_Vulnerability_Scoring_System

**Comment by Sjoske** [comment]
I suggest that it isn't possible to use a word, cause they can use standard words like flamingo, beheerder or administrator

**Comment by mprins** [comment]
Either https://github.com/nulab/zxcvbn4j or https://github.com/GoSimpleLLC/nbvcxz will check password "strength" against one or more dictionaries with well-known "words"

**Comment by mtoonen** [comment]
> Note that the current password policy (length/complexity) has received a score of "CVSS:3.0/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:N" in recent a penetration test, flagging this as a "LOW" severity vulnerabilty, so I think something should be done
> 
> see:
> 
> * https://nvd.nist.gov/vuln-metrics/cvss
> * https://en.wikipedia.org/wiki/Common_Vulnerability_Scoring_System

Agreed. Setting the minimum length to a higher value seems enough, and don't require extra libraries/coding, so lets start with that

**Comment by mprins** [comment]
het zou goed zijn als zowel de validatie als het hashing algoritme pluggable zouden zijn (zoals ook de audit module pluggable is) zodat ze makkelijk geupgraded of vervangen kunnen worden, onafhankelijk van de JPA code
